### PR TITLE
notification: improve password reset notifications

### LINKED ISSUE #1387: Reset password e-mails are to short and not translated
**Describe the bug**

The reset password e-mails (reset + confirmation) are not or partially translated.

**To Reproduce**
1. Ask to reset your password
1. See e-mail received
1. Reset the password
1. See confirmation

**Expected behavior**
The content is trnaslated, more exhaustive and nicer! See below

```
SUBJECT: RERO ID password reset

Dear [first name and last name],

Someone requested that the password for your RERO ID account be reset.

Reset my password(+link)

If you didn't request this, you can ignore this e-mail. Your password won't change until you create a new password.

Best regards

Your library
http:://ils.test.rero.ch/(adapt this link according to the instance)

```
Template: https://postmarkapp.com/blog/password-reset-templates-best-practices

```
SUBJECT: Your RERO ID password has been reset

Dear [first name and last name],

Your password has been successfully reset.

Best regards

Your library
http:://ils.test.rero.ch/(adapt this link according to the instance)
```

**Context**
* server: ilspilot.test.rero.ch

Proposed by the pilot libraries

**Screenshots**

Reset
image

Confirmation
image

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Improves reset instruction notification.
* Adds a custom template for reset notice notification.
* Closes #1387.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>

## Why are you opening this PR?

To solve #1387.

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

N/A

## How to test?

Go to login page and ask to change the password of a user. The template of the emails are logged on the server console.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by zannkukai — APPROVED** [review]

File: rero_ils/filter.py [hunk 1]
```diff
@@ -134,3 +136,18 @@ def empty_data(data, replacement_string='No data'):
...
+def lib_url(user):
+    """Return organisation viewcode.
+
+    :param user - invenio user to get viewcode
+    """
+    viewcode = current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE')
+    patron = Patron.get_patron_by_user(user)
+    if patron is not None:
+        viewcode = patron.get_organisation().get('code')
+    return '{base_url}/{viewcode}'.format(
```
Comment (jma) [inline rero_ils/filter.py:150]: If it is possible try to use `url_for`, but I'm not sure that it is available.

File: rero_ils/modules/ext.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def __init__(self, app=None):
             # it is require for Flask-Security see:
             # https://pythonhosted.org/Flask-Security/customizing.html#emails
             ils_loader = jinja2.ChoiceLoader([
-                jinja2.PackageLoader('rero_ils', 'templates'),
+                jinja2.PackageLoader('rero_ils', 'theme/templates'),
```
Comment (jma) [inline rero_ils/modules/ext.py:66]: Should have a mention in the commit message.

File: rero_ils/theme/templates/security/email/reset_instructions.html [hunk 3]
```diff
@@ -18,4 +18,14 @@
 
 #}
 
-<p><a href="{{ reset_link.replace('/api', '') }}">{{ _('Click here to reset your password') }}</a></p>
\ No newline at end of file
+{%- if user.profile.first_name and user.profile.last_name %}
+<p>{{ _('Dear') }} {{ user.profile.first_name }} {{ user.profile.last_name }},</p>
+{%- else %}
+<p>{{ _('Dear patron') }},</p>
```
Comment (jma) [inline rero_ils/theme/templates/security/email/reset_instructions.html:24]: Is this has been validate by the PO? Sounds wired to me.

File: rero_ils/theme/templates/security/email/reset_instructions.txt [hunk 4]
```diff
@@ -1,3 +1,12 @@
-{{ _('Click the link below to reset your password:') }}
 
-{{ reset_link.replace('/api', '') }}
\ No newline at end of file
+{%- if user.profile.first_name and user.profile.last_name %}
+<p>{{ _('Dear') }} {{ user.profile.first_name }} {{ user.profile.last_name }},</p>
+{%- else %}
+<p>{{ _('Dear patron') }},</p>
```
Comment (jma) [inline rero_ils/theme/templates/security/email/reset_instructions.txt:5]: Same as previous.

File: rero_ils/theme/templates/security/email/reset_notice.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#}
+
+{%- if user.profile.first_name and user.profile.last_name %}
+<p>{{ _('Dear') }} {{ user.profile.first_name }} {{ user.profile.last_name }},</p>
+{%- else %}
+<p>{{ _('Dear patron') }},</p>
+{%- endif %}
+<p>{{ _('Your password has been successfully reset.') }}</p>
```
Comment (jma) [inline rero_ils/theme/templates/security/email/reset_notice.html:26]: TODO: in a futur, we need a message to explain what to do if the patron did not change his password. (Hacking)

File: rero_ils/theme/templates/security/email/reset_notice.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{%- if user.profile.first_name and user.profile.last_name %}
+<p>{{ _('Dear') }} {{ user.profile.first_name }} {{ user.profile.last_name }},</p>
+{%- else %}
+<p>{{ _('Dear patron') }},</p>
+{%- endif %}
+<p>{{ _('Your password has been successfully reset.') }}</p>
```
Comment (jma) [inline rero_ils/theme/templates/security/email/reset_notice.txt:6]: TODO: in a futur, we need a message to explain what to do if the patron did not change his password. (Hacking)

**Review by jma — APPROVED** [review]

File: rero_ils/filter.py (hunk 1 above)
Comment (AoNoOokami) [reply rero_ils/filter.py:150]: No, I tried with url_for but it's not available here.

File: rero_ils/theme/templates/security/email/reset_instructions.html (hunk 3 above)
Comment (AoNoOokami) [reply rero_ils/theme/templates/security/email/reset_instructions.html:24]: Yes

**Pull request merged**